bokeh/charts/glyphs.py

```diff
@@ -912,13 +913,29 @@ class HistogramGlyph(AggregateGlyph):
 
     # input properties
     bin_width = Float()
-    bin_count = Float(help="""Provide a manually specified number of bins to use.""")
+    bin_count = Either(List(Float), Int, default=None, help="""
+    If bins is an int, it defines the number of equal-width bins in the
+    given range (10, by default). If bins is a sequence, it defines the
+    bin edges, including the rightmost edge, allowing for non-uniform
+    bin widths.
+
+    (default: None, use Freedman-Diaconis rule)
+    """)
```

Comment:
@bryevdv right.. At some point I was renaming `bin_count` to `bins` that is really the best name (just like the `np.histogram` argument) to then realize that there is some work to be done to avoide clashing names (HistogramGlyph has already a property `bins` that actually could be just turned to a `_bins` attribute but then it'd also scaled down to `Stats` that has the same problem... My take was to rename those internal props with some more thinking in a following PR (since it won't break exposed Charts API because `bins` is the name we expose right now). 

That's said I'm ok with doing it in this PR (was trying to switch to a more "small and fast" PRs mindset :) )